File: zapp-common/src/main/java/com/zfoo/app/zapp/common/protocol/group/model/GroupMemberSimpleVO.java
/*
 * Copyright (C) 2020 The zfoo Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and limitations under the License.
 */

package com.zfoo.app.zapp.common.protocol.group.model;

import com.zfoo.protocol.IPacket;

import java.util.List;

/**
 * @author jaysunxiao
 * @version 1.0
 * @since 2020-05-04 19:59
 */
public class GroupMemberSimpleVO implements IPacket {

    public static final transient short PROTOCOL_ID = 18013;

    private long groupId;

    private long memberId;

    private GroupTimeVO groupTime;

    private List<Long> groupAuthIds;

    public static GroupMemberSimpleVO valueOf(long groupId, long memberId, GroupTimeVO groupTime, List<Long> groupAuthIds) {
        var vo = new GroupMemberSimpleVO();
        vo.groupId = groupId;
        vo.memberId = memberId;
        vo.groupTime = groupTime;
        vo.groupAuthIds = groupAuthIds;
        return vo;
    }

    @Override
    public short protocolId() {
        return PROTOCOL_ID;
    }

    public long getGroupId() {
        return groupId;
    }

    public void setGroupId(long groupId) {
        this.groupId = groupId;
    }

    public long getMemberId() {
        return memberId;
    }

    public void setMemberId(long memberId) {
        this.memberId = memberId;
    }

    public GroupTimeVO getGroupTime() {
        return groupTime;
    }

    public void setGroupTime(GroupTimeVO groupTime) {
        this.groupTime = groupTime;
    }

    public List<Long> getGroupAuthIds() {
        return groupAuthIds;
    }

    public void setGroupAuthIds(List<Long> groupAuthIds) {
        this.groupAuthIds = groupAuthIds;
    }
}

File: zapp-gateway/jsProtocol/group/model/GroupMemberSimpleVO.js
// @author jaysunxiao
// @version 1.0
// @since 2020-05-04 19:59
const GroupMemberSimpleVO = function(groupAuthIds, groupId, groupTime, memberId) {
    this.groupAuthIds = groupAuthIds; // java.util.List<java.lang.Long>
    this.groupId = groupId; // long
    this.groupTime = groupTime; // com.zfoo.app.zapp.common.protocol.group.model.GroupTimeVO
    this.memberId = memberId; // long
};

GroupMemberSimpleVO.prototype.protocolId = function() {
    return 18013;
};

GroupMemberSimpleVO.write = function(buffer, packet) {
    if (buffer.writePacketFlag(packet)) {
        return;
    }
    buffer.writeLongArray(packet.groupAuthIds);
    buffer.writeLong(packet.groupId);
    buffer.writePacket(packet.groupTime, 18014);
    buffer.writeLong(packet.memberId);
};

GroupMemberSimpleVO.read = function(buffer) {
    if (!buffer.readBoolean()) {
        return null;
    }
    const packet = new GroupMemberSimpleVO();
    const list0 = buffer.readLongArray();
    packet.groupAuthIds = list0;
    const result1 = buffer.readLong();
    packet.groupId = result1;
    const result2 = buffer.readPacket(18014);
    packet.groupTime = result2;
    const result3 = buffer.readLong();
    packet.memberId = result3;
    return packet;
};

export default GroupMemberSimpleVO;
